FirebaseAuth/Sources/Storage/FIRAuthKeychainServices.m

```diff
@@ -226,11 +226,28 @@ - (void)deleteLegacyItemWithKey:(NSString *)key {
     @param key The key for the value being manipulated, used as the account field in the query.
  */
 - (NSDictionary *)genericPasswordQueryWithKey:(NSString *)key {
-  return @{
+  NSMutableDictionary *query = @{
     (__bridge id)kSecClass : (__bridge id)kSecClassGenericPassword,
     (__bridge id)kSecAttrAccount : [kAccountPrefix stringByAppendingString:key],
     (__bridge id)kSecAttrService : _service,
-  };
+  }
+                                   .mutableCopy;
+
+  // TODO(ncooke3): Refactor Auth to provide a user defaults based
+  // implementation for unit testing purposes on macOS.
+#ifndef FIREBASE_AUTH_MACOS_TESTING
```

Comment:
The alternative to `#if`'ing out source code was to disable most of `FIRAuthTests.m`–– which seemed even worse.

Reply:
I think a long term solution would be to refactor for better dependency injection since accessing the keychain really isn't necessary for unit testing.